packages/integrations/mdx/src/index.ts

```diff
@@ -129,7 +129,7 @@ export default function mdx(partialMdxOptions: Partial<MdxOptions> = {}): AstroI
 										const compiled = await processor.process(vfile);
 
 										return {
-											code: escapeViteEnvReferences(String(compiled.value)),
+											code: String(compiled.value),
```

Comment:
We can safely remove this in a non breaking way as the plugin calls `fs.readFile`, bypassing Vite's transform. The only continuing env replacement left after this transform is Vite's define, which we know is fixed in Vite 5.

Luckily Astro's env replacements (which didn't work well in plain strings) happen before the `fs.readFile`, so the broken code was thrown away.